src/common/src/util/scan_range.rs

```diff
@@ -72,6 +72,69 @@ impl ScanRange {
         }
     }
 
+    /// `Precondition`: make sure the first order key is int if you call this method.
+    /// Optimize small range scan. It turns x between 0 and 5 into x in (0, 1, 2, 3, 4, 5).
+    pub fn split_small_range(&self) -> Option<Vec<Self>> {
+        if self.eq_conds.is_empty() {
+            match self.range {
+                (
+                    Bound::Included(ScalarImpl::Int16(ref left)),
+                    Bound::Included(ScalarImpl::Int16(ref right)),
+                ) => {
+                    let gap = right - left + 1;
+                    if gap <= 8 {
```

Comment:
As we construct 256 iterators for unknown-vnode range scan, I guess this will benefit to an even larger gap. Is there any explanations for using 8 here?